### PR TITLE
fix(cli): panic on wait command if event is null

### PR DESCRIPTION
**Pull request by sarabala1979**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: cmd/argo/commands/wait.go [hunk 1]
```diff
@@ -91,8 +91,12 @@ func waitOnOne(serviceClient workflowpkg.WorkflowServiceClient, ctx context.Cont
 			log.Debug("Re-establishing workflow watch")
 			stream, err = serviceClient.WatchWorkflows(ctx, req)
 			errors.CheckError(err)
+			continue
 		}
 		errors.CheckError(err)
+		if event == nil {
+			continue
```
Comment (alexec) [inline cmd/argo/commands/wait.go:98]: why is event nil?

Comment (sarabala1979) [reply cmd/argo/commands/wait.go:98]: This is an extra check based on stack trace event is nil for panic.  Based on code, if re-establishment happens, an event may be nil.

**Comment by sarabala1979** [comment]
sure

**Comment by sarabala1979** [comment]
The fix has been backported to 2.9

**Pull request merged**